### ISSUE TITLE
False positive on type comparison

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#5199]
### Bug description

```python
cert_pub_key = cert.public_key()
key_pub_key = key.public_key()
if type(cert_pub_key) != type(key_pub_key) or cert_pub_key.public_numbers() != key_pub_key.public_numbers():
   xxx
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
C0123: Use isinstance() rather than type() for a typecheck. (unidiomatic-typecheck)
```

### Expected behavior

no error on the line.

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.9.7 (default, Aug 30 2021, 00:00:00) 
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
```

### OS / Environment

Fedora 34
